kolibri/plugins/user/assets/src/actions.js

```diff
@@ -22,6 +23,60 @@ function showRoot(store) {
   });
 }
 
+function editProfile(store, edits, session) {
+  // payload needs username, fullname, and facility
+  // used to save changes to API
+  const savedUserModel = FacilityUserResource.getModel(session.user_id);
+  const changedValues = {};
+
+  // TODO set up core session updates
+
+  // explicit checks for the only values that can be changed
+  if (edits.full_name && edits.full_name !== session.full_name) {
+    changedValues.full_name = edits.full_name;
+  }
+  if (edits.username && edits.username !== session.username) {
+    changedValues.username = edits.username;
+  }
+  if (edits.password && edits.password !== session.password) {
+    changedValues.password = edits.password;
+  }
+
+  // check to see if anything's changed and conditionally add last requirement
+  if (Object.keys(changedValues).length) {
+    changedValues.facility = session.facility_id;
+  } else {
+    return;
+  }
+
+  // update user object with new values
+  store.dispatch('SET_PROFILE_BUSY', true);
+
+  savedUserModel.save(changedValues).then(userWithAttrs => {
```

Comment:
this is very minor, but I've been trying to enforce a whitespace convention with promises where all function parameters are indented consistently, like this: https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/actions.js#L112

e.g.:

```javascript
  savedUserModel.save(changedValues).then(
    userWithAttrs => {
      // dispatch changes to store
      coreActions.getCurrentSession(store);
      store.dispatch('SET_PROFILE_SUCCESS', true);
      store.dispatch('SET_PROFILE_BUSY', false);
      store.dispatch('SET_PROFILE_EROR', false, '');
    },
    // error handling
    error => {
      // copying logic from user-create-modal
      function errorMessage(apiError) {
        if (apiError.status.code === 400) {
          // access the first apiError message
          return Object.values(apiError.entity)[0][0];
        } else if (apiError.status.code === 403) {
          return apiError.entity[0];
        }
        return '';
      }
      store.dispatch('SET_PROFILE_SUCCESS', false);
      store.dispatch('SET_PROFILE_EROR', true, errorMessage(error));
      store.dispatch('SET_PROFILE_BUSY', false);
    }
  );
}
```